js/main.js

```diff
@@ -379,12 +380,16 @@ $(document).ready(function() {
     legend.onAdd = function () { return L.DomUtil.get('legend'); };
 
     // Populate dropdown
-    loadCityIDs().done(function(cityIDs) {
-        for (var i = 0; i < cityIDs.length; i++) {
-            var id = cityIDs[i];
+    loadCities().done(function(cities) {
+
+        // cache the results
+        cityDirectory = {};
+        for (var i = 0; i < cities.length; i++) {
```

Comment:
Please don't. Caching `.length` has minimal performance advantages at best and [can even be slower](http://stackoverflow.com/a/5752991/857390). In any case it does not justify the reduction in readability.